docs/architecture/tablets.rst

```diff
@@ -0,0 +1,99 @@
+=========================================
+Data Distribution with Tablets
+=========================================
+
+A ScyllaDB cluster is a group of interconnected nodes - on-premises servers or 
+virtual servers (public cloud instances). The data of the entire cluster has to
+be distributed as evenly as possible across those nodes.
+
+ScyllaDB is designed to ensure a balanced distribution of data by storing data
+in tablets. When you add or remove nodes to scale your cluster, add or remove
+a datacenter, or replace a node, tablets are moved between the nodes to keep
+the same number on each node.
+
+This article explains the concept of tablets and how they let you scale your
+cluster quickly and seamlessly.
+
+Data Distribution
+-------------------
+
+ScyllaDB distributes data by splitting tables into tablets. Each tablet has 
+its replicas on different nodes, depending on the RF (replication factor). Each
+partition of a table is mapped to a single tablet in a deterministic way. When you
+query or update the data, ScyllaDB can quickly identify the tablet that stores
+the relevant partition. 
+
+The following example shows a 3-node cluster with a replication factor (RF) of
+3. The data is stored in a table (Table 1) with two rows. Both rows are mapped
+to one tablet (T1) with replicas on all three nodes.
+
+.. image:: images/tablets-cluster.png
+
+.. TODO - Add a section about tablet splitting when there are more triggers,
+   like throughput. In 6.0, tablets only split when reaching a threshold size
+   (measured in the number of rows).
+
+Load Balancing
+==================
+
+ScyllaDB autonomously moves tablets to balance the load based on table usage. 
+This process is managed by a load balancer mechanism and happens independently
+of the administrator or node operations such as bootstrap or decommission.
+The tablet load balancer decides where to migrate the tablets, either within
+the same node to balance the shards or across the nodes to balance the global
+load in the cluster. 
+
+The load-balancing process takes place in the background and is performed
+without any service interruption.
+
+Scaling Out
+=============
+
+A tablet can be dynamically migrated to either an existing node or a new node –
+on bootstrap. Tablets make scaling out fast and efficient, as the new node
+starts taking part of the load even before the bootstrap process completes.
+Also, tablets paired with consistent topology updates with Raft allow you to
+add multiple nodes simultaneously. Existing nodes stream data to the new ones,
+and the system load eventually converges to an even distribution as the process
+completes.
```

Comment:
It is important to mention in this context two other advantages of tablets-based scale-out vs. vnodes:
1. Streaming is more efficient as whole sstables are streamed so less data is read at the source (and in enterprise, file-based streaming is an additional improvement since it streams the data verbatim, without having to translate it to and from mutations over the wire).
2. Cleanup is automatic per-tablet so there is no need for manual cleanup (making tablets-based streaming safer), plus, since whole sstables are streamed, streaming is much cheaper as it involves mere deletion of said sstables, without having to rewrite the affected sstables as with vnodes (requiring significant amount of storage I/O and cpu processing).